pkg/gatherers/clusterconfig/operators_pods_and_events.go

```diff
@@ -313,6 +314,22 @@ func getContainerLogs(ctx context.Context,
 	return records
 }
 
+// getLogWithStacktracing search for the first stack trace line and offset it by logLinesOffset
+func getLogWithStacktracing(logArray []string) string {
+	var limit int
+	for idx := range logArray {
+		line := logArray[idx]
+		if found := stackTraceRegex.MatchString(line); found {
+			limit = idx - logLinesOffset
+			if limit < 0 {
+				limit = 0
+			}
+			break
+		}
+	}
+	return strings.Join(logArray[limit:], "\n")
```

Comment:
I think it looks good. Maybe we could copy the slice here to avoid keeping the underlying array in memory. I am leaving the final approval to @Sergey1011010 :-P

Reply:
Yeah. It also works, but I think that it is OK. The GC should handle it.

Reply:
Yeah, it should be removed since we're returning the result of `join`. Btw, if nothing is found, it will return all the logs, it is expected, right?

Reply:
Yep. It will return all the logs. But I think that would be really difficult to fit into this scenario, once we already detect the stack trace in the 100 last lines then we request the 2000 lines.

Reply:
Yes this function is called only when the regex pattern is found in the log snippet so there should be a line matching it as well.